src/main/java/org/opensearch/security/configuration/DlsFlsValveImpl.java

```diff
@@ -521,12 +521,30 @@ private void setFlsHeaders(EvaluatedDlsFlsConfig dlsFls, ActionRequest request)
                         }
                     }
                 } else {
-                    threadContext.putHeader(
-                        ConfigConstants.OPENDISTRO_SECURITY_MASKED_FIELD_HEADER,
-                        Base64Helper.serializeObject((Serializable) maskedFieldsMap)
-                    );
-                    if (log.isDebugEnabled()) {
-                        log.debug("attach masked fields info: {}", maskedFieldsMap);
+                    boolean shouldIncludeFlsHeaders = true;
+                    if (request instanceof SearchRequest) {
+                        SearchRequest searchRequest = (SearchRequest) request;
+                        boolean hasOnlyCountOrCardinality = true;
+                        if (searchRequest.source().aggregations() != null) {
+                            for (AggregationBuilder af : searchRequest.source().aggregations().getAggregatorFactories()) {
+                                if (!(af.getType().equals("cardinality") || af.getType().equals("count"))) {
+                                    hasOnlyCountOrCardinality = false;
```

Comment:
should it be not set to true once its set to false? If so you can add a `break;` here to skip the rest of the loop.